Parse/src/test/java/com/parse/ParseUserTest.java

```diff
@@ -1059,6 +1059,25 @@ public void testUpgradeToRevocableSessionAsync() throws Exception {
     verify(currentUserController, times(1)).setAsync(user);
   }
 
+  @Test
+  public void testDontOverwriteSessionTokenForCurrentUser() throws Exception {
+    ParseUser.State sessionTokenState = new ParseUser.State.Builder()
+            .sessionToken("sessionToken")
+            .build();
+    ParseUser.State emptyState = new ParseUser.State.Builder().build();
+
+    ParseUser user = ParseObject.from(sessionTokenState);
+    user.setIsCurrentUser(true);
+    assertEquals(user.getSessionToken(), "sessionToken");
+
+    user.setState(emptyState);
+    assertEquals(user.getSessionToken(), "sessionToken");
```

Comment:
Could we add some data in `emptyState` that we would want to merge in and verify that it's getting merged in correctly? A key that was in `sessionTokenState` that should be removed, a key that should be overwritten, and a key that should be added?